microprofile/WFLY-14940_upgrade-microprofile-health-3.1.adoc

```diff
@@ -0,0 +1,69 @@
+= Upgrade MicroProfile Health to 3.1
+:author:            Martin Stefanko
+:email:             mstefank@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,health,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 4.1 provides a minor upgrade for https://github.com/eclipse/microprofile-health/releases/tag/3.1[MicroProfile Health 3.1] with changes described in its https://download.eclipse.org/microprofile/microprofile-health-3.1/microprofile-health-spec-3.1.html#release_notes_3_1[Release Notes].
+
+This component upgrade also covers the upgrade of smallrye-health 3.1.1 that implements MicroProfile Health 3.1.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-14940[WFLY-14040]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1763[EAP7-1763]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[X] Engineering
+[ ] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 3.1
+* Upgrade SmallRye Health to 3.1.1
+* Pass all MicroProfile Health tests in WildFly integration test suite
+* Pass the MicroProfile Health 3.1 TCK without failures
+* Adjust the tests to work as needed
+* Adjust QuickStarts to showcase new Startup feature
+
+
+=== Non-Requirements
+
+== Implementation Plan
+The MicroProfile Health 3.1 is an incremental, backwards compatible release that only added one major feature called Startup probes (@Startup qualifier and /health/startup endpoint) for the Kubernetes startup probes.
+
+== Test Plan
+
+The implementation should pass the TCK and also add tests for the mp.health.default.startup.empty.response as this functionality cannot be implemented in the TCK itself.
+
+* Run the WildFly integration microprofile test suite and checks there are no failures related to MicroProfile Health tests.
+* Run the `microprofile-tck/health` module from WildFly integration test suite and check that there are no failures.
+* Add additional tests for mp.health.default.startup.empty.response
+
+== Community Documentation
+
+* No community documentation is required for this upgrade (covered by the specification)
+
+== Breaking changes
+
+There were no breaking changes.
```

Comment:
The `Release Notes` [section](https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc#release-note-content) is missing.
It should be here and provide text that would be used both for a WildFly release (e.g.: new etc.) or as a starting point for documentation drafts for the product.

Reply:
Can be resolved (I don't have the grants for doing so), thanks.